contracts/DomainRegistry.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+/// @title Domain Registry Contract
+/// @notice This contract allows users to register and manage top-level domain names
+/// @author developed by Ivan Myasoyedov
+contract DomainRegistry {
+    /// @notice Address of the contract owner
+    address public owner;
+
+    /// @notice Fee required to register a domain
+    uint256 public registrationFee;
+
+    /// @dev Struct to store information about a domain
+    struct Domain {
+        string name;
+        address controller;
+        uint256 registeredAt;
+    }
+
+    /// @dev Array of all domains registered in the contract
+    Domain[] public domains;
+
+    /// @dev Mapping from domain name to controller address
+    mapping(string => address) public domainToController;
+
+    /// @dev Mapping from controller address to their domains
+    mapping(address => Domain[]) public controllerToDomains;
+
+    /// @notice Emitted when a new domain is registered
+    event DomainRegistered(
+        string domain,
+        address indexed controller,
+        uint256 timestamp
+    );
+
+    /// @dev Ensures that only the owner can call the function
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Only the owner can call this function");
+        _;
+    }
+
+    /// @notice Creates a new DomainRegistry contract
+    /// @param _registrationFee The fee required to register a new domain
+    constructor(uint256 _registrationFee) {
+        owner = msg.sender;
+        registrationFee = _registrationFee;
+    }
+
+    /// @notice Retrieves the domains controlled by a specific address
+    /// @param _controllerAddress The address of the controller to query domains for
+    /// @return An array of `Domain` structs controlled by `_controllerAddress`
+    function getControllerDomains(address _controllerAddress) public view returns (Domain[] memory) {
```

Comment:
line 53 
function may be external and save some gas
when you return array as value with small length it will work, but if the array is big it will not work
function should be moved according to NatSpec rules
// - 1 point